File: 02_Lua/Psuedocode/AddTwoNums.lua
--print("Hello from AddTwoNums")
--take two numbers from user, add together, print result 
function printTwoNumSum()
    --variables
    local num1 = nil
    local num2 = nil
    local sum = 0

    while (num1 == nil or num1 == "" or not tonumber(num1)) do
        print("What is your first number: ")
        num1 = io.read()
    end

    while (num2 == nil or num2 == "" or not tonumber(num2)) do
        print("What is your second number: ")
        num2 = io.read()
    end
    sum = num1 + num2

    print(sum)
end

File: 03_Python/Psuedocode/AddTwoNums.py
#print("Hello from addTwo")
#take two numbers from user and print the sum


#Is the input a number?
def IsNum(userInput):
    return bool(userInput.isdigit() or isinstance(userInput, float))

#While not a number, keep prompting
def TryAgain(userInput):
    while not IsNum(userInput):
        userInput = input("That's not a number I can add; try again: ")
    return userInput


num1 = input("Give me a number: ")
num1 = TryAgain(num1)
num2 = input("Give me another number: ")
num2 = TryAgain(num2)
print(num1 + " + " + num2 + " = " + str((float(num1) + float(num2))))